packages/logger/src/utils/format.ts

```diff
@@ -87,7 +88,6 @@ function humanReadableTemplateFn(_info: {[key: string]: any; level: string; mess
   str += `[${infoString}] ${info.level.padStart(infoPad)}: ${info.message}`;
 
   if (info.context !== undefined) str += " " + logCtxToString(info.context);
-  if (info.error !== undefined) str += " " + logCtxToString(info.error);
-
+  if (info.error !== undefined) str += (info.error instanceof LodestarError ? " " : " - ") + logCtxToString(info.error);
```

Comment:
Padding ` - ` for an `Error` class and not for `LodestarError` seems not so informative. 

If you want to process different errors differently would prefer to chance the logic of `logCtxToString`, as there is already custom logic based on error constructor.

Reply:
> Padding - for an Error class and not for LodestarError seems not so informative.

You are right, I am not happy with this check either that's why I explicitly also added a comment in the PR.

Regarding your suggestions to move it to `logCtxToString`, this would be ideal but is problematic as well
- `logCtxToString` calls itself recursively
https://github.com/ChainSafe/lodestar/blob/13978344a312bd5238e11f26839016331a382799/packages/logger/src/utils/json.ts#L123
- main purpose of this function is to convert a context to string and not handle log message formatting

I am also not sure if LodestarError is ever passed like this to log function, the only place I can see in the code (outside of tests) is here
https://github.com/ChainSafe/lodestar/blob/13978344a312bd5238e11f26839016331a382799/packages/beacon-node/src/util/error.ts#L48

I just added this extra differentiation to make the test here happy
https://github.com/ChainSafe/lodestar/blob/13978344a312bd5238e11f26839016331a382799/packages/logger/test/fixtures/loggerFormats.ts#L34-L45

but never saw this error in production

I see three options
1. use " - " separator in all cases and update test case
2. use " - " separator in all cases and remove test case
3. differentiate based on Error vs LodestarError (current approach)

Might be best to just go with (1) but let me know what you think

Reply:
Yes I would suggest to go with option 1.